tools/rc/ResourceCompiler.cpp

```diff
@@ -110,6 +112,178 @@ std::string us_tempfile()
+};
+
+/*
+ * @brief parses json content and returns the parsed json or throws.
+ * @tparam jsonContent json content to parse. The type must be convertible to std::istream.
+ * @param root The parsed Json root object.
+ * @throw InvalidManifest if the json is invalid. Parse error information is in the exception.
+ * If an exception is thrown, the root param is invalid.
+ */
+template <class T>
+void parseAndValidateJson(T& jsonContent, Json::Value& root)
+{
+  Json::CharReaderBuilder rbuilder;
+  rbuilder["rejectDupKeys"] = true;
+  rbuilder["allowComments"] = false;
+  std::string errs;
+
+  if (!Json::parseFromStream(rbuilder, jsonContent, &root, &errs))
+  {
+    throw InvalidManifest(errs);
+  }
+}
+
+/*
+ * @brief parses json content from a file and returns the parsed json or throws.
+ * @param jsonFile path to a json file.
+ * @param root The parsed Json root object.
+ * @throw InvalidManifest if the json is invalid. Parse error information is in the exception.
+ * If an exception is thrown, the root param is invalid.
+ */
+void parseAndValidateJsonFromFile(const std::string& jsonFile, Json::Value& root)
+{
+  try
+  {
+    std::ifstream json(jsonFile);
+    parseAndValidateJson(json, root);
+  }
+  catch (const InvalidManifest& e)
+  {
+    std::string exceptionMsg(jsonFile + ": " + e.what());
+    throw InvalidManifest(exceptionMsg);
+  }
+}
+
+/* 
+ * @brief Validate manifest files in an archive.
+ * @param archiveFile archive file path
+ * @throw std::InvalidManifest on the first invalid manifest found.
+ */
+void validateManifestsInArchive(const std::string archiveFile)
+{
+  mz_zip_archive currZipArchive;
+  mz_uint currZipIndex = 0;
+  memset(&currZipArchive, 0, sizeof(mz_zip_archive));
+  std::clog << "Validating manifests in " << archiveFile << " ... " << std::endl;
+  if (!mz_zip_reader_init_file(&currZipArchive, archiveFile.c_str(), 0))
+  {
+    throw std::runtime_error("Could not initialize zip archive " + archiveFile);
+  }
+  char archiveName[MZ_ZIP_MAX_ARCHIVE_FILENAME_SIZE];
+  mz_uint numZipIndices = mz_zip_reader_get_num_files(&currZipArchive);
+  for (currZipIndex = 0; currZipIndex < numZipIndices; ++currZipIndex)
+  {
+    mz_uint numBytes = mz_zip_reader_get_filename(&currZipArchive, currZipIndex, archiveName, sizeof archiveName);
+    std::clog << "\tValidating: " << archiveName << " (from " << archiveFile << ") "<< std::endl;
+    try
+    {
+      if (numBytes > 1 && archiveName[numBytes-2] != '/') // The last character is '\0' in the array
+      {
+        std::string archiveEntry(archiveName);
+        if (archiveEntry.find_first_of("/", 0) == archiveEntry.find_last_of("/") &&
+            std::string::npos != archiveEntry.find("/manifest.json"))
+        {
+          char manifestFileContents[MZ_ZIP_MAX_IO_BUF_SIZE];
+          memset(manifestFileContents, 0, MZ_ZIP_MAX_IO_BUF_SIZE);
+          if(MZ_TRUE == mz_zip_reader_extract_file_to_mem(&currZipArchive, 
```

Comment:
If I read this correctly, the `mz_zip_reader_extract_file_to_mem` call will fail for files with an uncompressed size `> MZ_ZIP_MAX_IO_BUF_SIZE` which is defined to 64 KB. This doesn't look right to me.

Reply:
good catch. This should be using ```mz_zip_reader_extract_to_heap``` instead.